code/ZenValidatorConstraint.php

```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
+class Constraint_dimension extends ZenValidatorConstraint
+{
+    protected $type;
+    protected $val1;
+    protected $val2;
+
+    const WIDTH = 'width';
+    const HEIGHT = 'height';
+    const WIDTH_HEIGHT = 'width_height';
+    const RATIO = 'ratio';
+    const MIN_WIDTH = 'min_width';
+    const MIN_HEIGHT = 'min_height';
+    const MIN_WIDTH_HEIGHT = 'min_width_height';
+    const MAX_WIDTH = 'max_width';
+    const MAX_HEIGHT = 'max_height';
+    const MAX_WIDTH_HEIGHT = 'max_width_height';
+
+    public function __construct($type,$val1,$val2=null)
+    {
+        $this->type = $type;
+        $this->val1 = $val1;
+        $this->val2 = $val2;
+
+        parent::__construct();
+    }
+
+    /**
+     * Validate function called for validator.
+     * @param  Mixed $value the value of the field being validated.
+     * @return boolean
+     */
+    public function validate($value)
+    {
+        // The value which comes in is a files array so we can look through this
+        // to and then get the files to test aspects of them.
+        if (isset($value['Files'])) {
+            foreach($value['Files'] as $fileID) {
+                $file = File::get()->byId($fileID);
+
+                // Now have the file double-check it is an image and if so then
+                // get some information about the image using PHPs getimagesize()
+                if ($file->ClassName == 'Image') {
+                    $info = getimagesize($_SERVER['DOCUMENT_ROOT'] . "/" . $file->Filename);
+
+                    if ($info && is_array($info)) {
+                        $width = $info[0];
+                        $height = $info[1];
+
+                        switch ($this->type) {
+                            case self::WIDTH:
+                                return $width == $this->val1;
+                                break;
+                            case self::HEIGHT:
+                                return $height == $this->val1;
+                                break;
+                            case self::WIDTH_HEIGHT:
+                                if (($width == $this->val1) && ($height == $this->val2)) {
+                                    return true;
+                                } else {
+                                    return false;
+                                }
+                                break;
+                            case self::RATIO:
+                                return $this->ratioValidation($width, $height);
+                                break;
+                            case self::MIN_WIDTH:
+                                return $width >= $this->val1;
+                                break;
+                            case self::MIN_HEIGHT:
+                                return $height >= $this->val1;
+                                break;
+                            case self::MIN_WIDTH_HEIGHT:
+                                if (($width >= $this->val1) && ($height >= $this->val2)) {
```

Comment:
Could be simplified to
   return (($width >= $this->val1) && ($height >= $this->val2));